### PR TITLE
[misc] Fixing the migration of StaticListClassFieldTest to a docker integration test

### PR DESCRIPTION
**Pull request by manuelleduc**
- Add testMultipleSelect inherited from AbstractListClassFieldTest to StaticListClassFieldIT
- Move isReadOnly up to ListClassFieldEditPane
- Move testMultipleSelect to DBListClassFieldTest
- Use ListClassFieldEditPane#isReadOnly in DBListClassFieldTest#testMultipleSelect
- Remove the now unused AbstractListClassFieldTest

### CODE REVIEW

File: xwiki-platform-distribution/xwiki-platform-distribution-flavor/xwiki-platform-distribution-flavor-test/xwiki-platform-distribution-flavor-test-ui/src/test/it/org/xwiki/test/ui/appwithinminutes/DBListClassFieldTest.java [hunk 1]
```diff
@@ -75,10 +73,50 @@ public void testDisplayType()
...
     }
+
+    /**
+     * Tests that multiple select state is synchronized with the rest of the meta properties.
+     *
+     * @since 13.2
+     * @since 12.10.6
+     */
+    @Test
+    public void testMultipleSelect()
```
Comment (surli) [inline xwiki-platform-distribution/xwiki-platform-distribution-flavor/xwiki-platform-distribution-flavor-test/xwiki-platform-distribution-flavor-test-ui/src/test/it/org/xwiki/test/ui/appwithinminutes/DBListClassFieldTest.java:90]: no public here and no "test" prefix no?

Comment (manuelleduc) [reply xwiki-platform-distribution/xwiki-platform-distribution-flavor/xwiki-platform-distribution-flavor-test/xwiki-platform-distribution-flavor-test-ui/src/test/it/org/xwiki/test/ui/appwithinminutes/DBListClassFieldTest.java:90]: This is not a docker test.
- with junit 4, the `public` visibility is mandatory
- I'm not that sure about the `test` prefix but looking at the rest of the module, it looks like the norm (I'd be happy to rename the methods of this test if it's not the case).

Comment (vmassol) [reply xwiki-platform-distribution/xwiki-platform-distribution-flavor/xwiki-platform-distribution-flavor-test/xwiki-platform-distribution-flavor-test-ui/src/test/it/org/xwiki/test/ui/appwithinminutes/DBListClassFieldTest.java:90]: > I'm not that sure about the test prefix but looking at the rest of the module, it looks like the norm (I'd be happy to rename the methods of this test if it's not the case).

It's not the norm (even if you see it a lot of places). See https://dev.xwiki.org/xwiki/bin/view/Community/Testing/JavaUnitTesting/#HBestpractices

> Name the test methods with the method to test followed by a qualifier describing the test. For example importWithHeterogeneousEncodings(). In the few cases where it's not applicable, name after the use case being tested.

this part is valid for both junit4 and junit5. When we introduced junit5, I removed it from the junit4 doc (I moved everything still valid to the junit5 doc).

Comment (manuelleduc) [reply xwiki-platform-distribution/xwiki-platform-distribution-flavor/xwiki-platform-distribution-flavor-test/xwiki-platform-distribution-flavor-test-ui/src/test/it/org/xwiki/test/ui/appwithinminutes/DBListClassFieldTest.java]: Thanks, I naively thought the rule was only holding for docker based tests. Knowledge and PR updated.

File: xwiki-platform-core/xwiki-platform-appwithinminutes/xwiki-platform-appwithinminutes-test/xwiki-platform-appwithinminutes-test-pageobjects/src/main/java/org/xwiki/appwithinminutes/test/po/ListClassFieldEditPane.java [hunk 2]
```diff
@@ -160,4 +160,14 @@ public String getPreviewInputType()
         }
         return null;
     }
+
+    /**
+     * @return {@code true} if the field is readonly, {@code false} otherwise
+     * @since 13.2
```
Comment (surli) [inline xwiki-platform-core/xwiki-platform-appwithinminutes/xwiki-platform-appwithinminutes-test/xwiki-platform-appwithinminutes-test-pageobjects/src/main/java/org/xwiki/appwithinminutes/test/po/ListClassFieldEditPane.java:166]: Note that we probably won't merge it on 13.2 branch now...

File: xwiki-platform-distribution/xwiki-platform-distribution-flavor/xwiki-platform-distribution-flavor-test/xwiki-platform-distribution-flavor-test-ui/src/test/it/org/xwiki/test/ui/appwithinminutes/DBListClassFieldTest.java [hunk 3]
```diff
@@ -75,10 +73,50 @@ public void testDisplayType()
...
         dbListField.closeConfigPanel();
         // Assert that the selected values have been preserved.
         assertEquals(Arrays.asList("AppWithinMinutes.DBList", "AppWithinMinutes.Date"),
             dbListField.getPicker().getValues());
     }
+
+    /**
+     * Tests that multiple select state is synchronized with the rest of the meta properties.
+     *
+     * @since 13.2
```
Comment (surli) [inline xwiki-platform-distribution/xwiki-platform-distribution-flavor/xwiki-platform-distribution-flavor-test/xwiki-platform-distribution-flavor-test-ui/src/test/it/org/xwiki/test/ui/appwithinminutes/DBListClassFieldTest.java:86]: Same here

**Pull request merged**